content.js

```diff
@@ -49,6 +55,49 @@ async function copyPrDescription() {
   messageFields.forEach(f => f.value = commitBody);
 }
 
+async function getPrMetadataFromDocument() {
+  await makePrDescriptionAppear();
+
+  const prTitleEl = document.getElementById('issue_title');
+  if (!prTitleEl) {
+    warn('failed to find PR title element');
+    return null;
+  }
+
+  let prBodyEl = document.querySelector('.comment-form-textarea[name="issue[body]"]');
+  if (!prBodyEl) {
+    prBodyEl = document.querySelector('.comment-form-textarea[name="pull_request[body]"]');
+    if (!prBodyEl) {
+      warn('failed to find PR body element');
+      return null;
+    }
+  }
+
+  return {
+    title: prTitleEl.value,
+    body: prBodyEl.textContent,
+  };
+}
+
+async function getPrMetadataFromApi(repo, prNumber) {
+  const response = await fetch(`https://api.github.com/repos/${repo}/pulls/${prNumber}`);
+  const prMetadata = await response.json();
+  return {
+    // These could be null in some cases, so replace nulls with empty strings.
+    title: prMetadata.title || '',
+    body: prMetadata.body || '',
+  };
+}
+
+async function makePrDescriptionAppear() {
```

Comment:
A side effect of doing this is that, at least on my computer, it scrolls to the top of the page which is a bit jarring.  I wonder if there's a way to either stop that from happening or to tell the browser to go back to the bottom of the page.

Makes me wonder if it wouldn't be better to start with API approach first, and save this for last since it is the most invasive to the user.  Plus, for public repos, the API approach seems to be very easy and future proof.  What do you think?

Reply:
Sure, that seems reasonable.

Reply:
Done